src/_nebari/stages/terraform_state/template/azure/modules/terraform-state/main.tf

```diff
@@ -4,9 +4,9 @@ resource "azurerm_resource_group" "terraform-state-resource-group" {
   tags     = var.tags
 }
 
+# DELETEME: This is where we create the storage account
 resource "azurerm_storage_account" "terraform-state-storage-account" {
-  # name, can only consist of lowercase letters and numbers, and must be between 3 and 24 characters long
-  name                     = replace("${var.name}${var.storage_account_postfix}", "-", "") # must be unique across the entire Azure service
```

Comment:
This was one of the causes of the bug.  The mutation logic where we create the terraform storage account is different from the mutation logic where we read the storage account.  I've moved the logic into the python layer so it's the same logic on creation of the storage account and reading of the storage account (terraform import).